api/src/main/java/org/apache/iceberg/catalog/Namespace.java

```diff
@@ -29,6 +31,8 @@
 public class Namespace {
   private static final Namespace EMPTY_NAMESPACE = new Namespace(new String[] {});
   private static final Joiner DOT = Joiner.on('.');
+  private static final Predicate<String> CONTAINS_NULL_BYTE =
+      Pattern.compile("\0|\u0000", Pattern.UNICODE_CHARACTER_CLASS).asPredicate();
```

Comment:
What's the difference between `\0` and `\u0000`?

Reply:
Using just `\0000` is sufficient, so I removed the ASCII one.